src/extensions/stores/workspace-store.ts

```diff
@@ -0,0 +1,109 @@
+import { Singleton } from "../core-api/utils";
+import { workspaceStore as internalWorkspaceStore, WorkspaceStore as InternalWorkspaceStore, Workspace, WorkspaceId } from "../../common/workspace-store";
+import { ObservableMap } from "mobx";
+
+export { Workspace } from "../../common/workspace-store";
+export type { WorkspaceId, WorkspaceModel } from "../../common/workspace-store";
+
+/**
+ * Stores all workspaces
+ */
+export class WorkspaceStore extends Singleton {
+  /**
+   * Default workspace id, this workspace is always present
+   */
+  static readonly defaultId: WorkspaceId = InternalWorkspaceStore.defaultId;
+
+  /**
+   * Currently active workspace id
+   */
+  get currentWorkspaceId(): string {
+    return internalWorkspaceStore.currentWorkspaceId;
+  }
```

Comment:
Do we not support extensions setting this?

Reply:
Good catch, added.